test/test_valgrind.sh

```diff
@@ -41,31 +49,45 @@ helgrind)
 	OPTION="--tool=helgrind"
 	;;
 *)
-	echo "error: unknown tool: $TOOL"
+	echo -e "error: unknown valgrind tool: $TOOL\n"
 	print_usage
 	exit 1
 	;;
 esac
 
-FAIL=0
+WORKSPACE=$(realpath $WORKSPACE)
+BUILD_DIR=$(realpath $BUILD_DIR)
 
+cd ${BUILD_DIR}/test/
 mkdir -p cpuid
 
 echo "Gathering data for hwloc so it can be run under valgrind:"
 hwloc-gather-cpuid ./cpuid
 
+echo
+echo "Working directory: $(pwd)"
 echo "Running: \"valgrind $OPTION\" for the following tests:"
 
-for tf in $(ls -1 ./test/umf_test-*); do
-	[ ! -x $tf ] && continue
-	echo -n "$tf "
-	LOG=${tf}.log
-	HWLOC_CPUID_PATH=./cpuid valgrind $OPTION $tf >$LOG 2>&1 || echo -n "(valgrind failed) "
-	if grep -q -e "ERROR SUMMARY: 0 errors from 0 contexts" $LOG; then
+FAIL=0
+rm -f *.log *.err
```

Comment:
I know it's just a build dir, but perhaps be more precise and only remove your logs, meaning `rm umf_test*.log umf_test*.err`

Reply:
Done